env.sh

```diff
@@ -1,71 +1,79 @@
+    CONTAINER_NAME="cita_run_container"
+    DOCKER_IMAGE="cita/cita-run:ubuntu-18.04-20181009"
+    SOURCE_DIR="$(dirname $SOURCE_DIR)"
 fi
 
-docker_bin=$(which docker)
-if [ -z "${docker_bin}" ]; then
-    echo "Command not found, install docker first."
-    exit 1
+if [ `uname` == 'Darwin' ]; then
+    cp /etc/localtime ${SOURCE_DIR}/localtime
+    SYSTEM_NET="bridge"
+    LOCALTIME_PATH="${SOURCE_DIR}/localtime"
 else
-    docker version > /dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        echo "Run docker version failed, Maybe docker service not running or current user not in docker user group."
-        exit 2
-    fi
+    SYSTEM_NET="host"
+    LOCALTIME_PATH="/etc/localtime"
 fi
 
-SOURCE_DIR=`pwd`
-CONTAINER_NAME="cita_build${SOURCE_DIR//\//_}"
-CARGO_HOME=/opt/.cargo
 WORKDIR=/opt/cita
 USER_ID=`id -u $USER`
 USER_NAME="user"
 
-if [ "${USER_ID}" = "0" ]; then
-    USER_NAME="root"
-fi
+[[ "${USER_ID}" = "0" ]] && USER_NAME="root"
 
-mkdir -p ${HOME}/.docker_cargo/git
-mkdir -p ${HOME}/.docker_cargo/registry
+# Init contanier's cargo, for logs.
+CARGO_HOME=/opt/.cargo
+DOCKER_CARGO=${HOME}/.docker_cargo
+mkdir -p ${DOCKER_CARGO}/git
+
+# Port condition
+EXPOSE="1337:1337"
+if [ "$3" == "port" ]; then
+    EXPOSE=${@:4}
+    [[ "${EXPOSE}" = "NULL" ]] && EXPOSE="1337:1337"
+    docker container stop $CONTAINER_NAME > /dev/null 2>&1
+fi
 
+# Run container
 docker ps | grep ${CONTAINER_NAME} > /dev/null 2>&1
 if [ $? -ne 0 ]; then
     echo "Start docker container ${CONTAINER_NAME} ..."
     docker rm ${CONTAINER_NAME} > /dev/null 2>&1
 
-    # test network and set init cmd
-    timeout=3
-    target=www.google.com
-    ret_code=`curl -I -s --connect-timeout $timeout $target -w %{http_code} | tail -n1`
-    if [ "x$ret_code" = "x200" ]; then
-        INIT_CMD="while true;do sleep 100;done"
-    else
-        INIT_CMD="echo -e '[source.crates-io]\nregistry = \"https://github.com/rust-lang/crates.io-index\"\nreplace-with = \"ustc\"\n[source.ustc]\nregistry = \"https://mirrors.ustc.edu.cn/crates.io-index\"' | sudo tee /opt/.cargo/config;while true;do sleep 100;done"
-    fi
-
     docker run -d \
+           --net=${SYSTEM_NET} \
            --volume ${SOURCE_DIR}:${WORKDIR} \
-           --volume ${HOME}/.docker_cargo/registry:${CARGO_HOME}/registry \
-           --volume ${HOME}/.docker_cargo/git:${CARGO_HOME}/git \
+           --volume ${DOCKER_CARGO}/git:${DOCKER_CARGO}/git \
            --volume ${LOCALTIME_PATH}:/etc/localtime \
            --env USER_ID=${USER_ID} \
            --workdir ${WORKDIR} \
-           --name ${CONTAINER_NAME} ${DOCKER_IMAGE} \
-           /bin/bash -c "${INIT_CMD}"
+           --name ${CONTAINER_NAME} \
+           -p $EXPOSE ${DOCKER_IMAGE} \
+           /bin/bash -c "while true;do sleep 100;done"
     # Wait entrypoint.sh to finish
     sleep 3
 fi
 
 test -t 1 && USE_TTY="-t"
 
-if [ $# -gt 0 ]; then
+# Start nodes outside container directly
+# if `daemon.sh` just use in start node?
+# Remote Commands:
```

Comment:
Is `bin/cita start test-chain/0` right according to the comments?

Reply:
`start` is a special command, every `start` parameter inputs, cita will run daemon automatically. 

here is my latest log:
```
$ ./bin/cita start test-chain/0
cita receive parameters: start test-chain/0
env.sh recive paraemters: /home/travis/build/clearloop/cita/target/install/bin/cita-env bin/cita bebop --daemon start test-chain/0
error: exec: "/home/travis/build/clearloop/cita/target/install/bin/cita-env": stat /home/travis/build/clearloop/cita/target/install/bin/cita-env: no such file or directory
```
The `cita-env` did run, but the error still coms...XD

Reply:
```
+++ dirname bin/cita
++ cd bin
++ pwd -P
+ CITA_BIN=/home/kaikai/Work/cita/target/install/bin
++ dirname /home/kaikai/Work/cita/target/install/bin
+ CITA_SCRIPTS=/home/kaikai/Work/cita/target/install/scripts
+ '[' start '!=' bebop ']'
+ '[' -e /home/kaikai/Work/cita/target/install/bin/cita-env ']'
+ /home/kaikai/Work/cita/target/install/bin/cita-env bin/cita bebop start test-chain/0
error: exec: "/home/kaikai/Work/cita/target/install/bin/cita-env": stat /home/kaikai/Work/cita/target/install/bin/cita-env: no such file or directory
```

Reply:
And, the `cita start` command in my branch, can run well while building locally or replaced them into the official compiled code, invoked into a formula.

Reply:
I think I might know what happened...I optimized the code in compiling scripts, but not update it in my branch.

Reply:
I raised the bug, after fixing `circles` —— messed up the `SOURCE_DIR`.